### DISCUSSION TITLE
Binary outcome, and multiple binary treatments.

### DISCUSSION THREAD

**Discussion by CJames1261** [DoubleML/doubleml-for-py#228] [Q&A] [answered]
Hi all,

I am currently trying to find the ATE of three binary treatments on a binary outcome. Which model would I use to do this? I know the DoubleMLIRM model can use binary treatments, but I can only use one binary treatment in the DoubleMLData d_cols paramter. Does anyone have a solution or any suggestions? Thanks in advance!

Here is my code: 
"""
import pandas as pd
import doubleml as dml
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split

' Load the dataset into a pandas dataframe'
df = pd.read_csv('/content/ai4i2020.csv')

'Assuming df is your DataFrame'
df['Machine failure'] = np.where((df['TWF'] == 1) | (df['HDF'] == 1) | (df['PWF'] == 1) | (df['OSF'] == 1) | (df['RNF'] == 1), 1, 0)

'Convert 'Machine failure' to integer'
df['Machine failure'] = df['Machine failure'].astype(int)

'One-hot encode categorical treatment variables'
df = pd.get_dummies(df, columns=['Type']).drop(['UDI', 'Product ID'], axis=1)

'Correct the data type assignments'
df['Type_H'] = df['Type_H'].astype(int)
df['Type_L'] = df['Type_L'].astype(int)
df['Type_M'] = df['Type_M'].astype(int)

'Define treatment, outcome, and covariates'
treatments = ['Type_M'] 
'##### I would like to use this: ['Type_M', 'Type_H', 'Type_L']#############'

outcome = 'Machine failure'
covariates = ['Air temperature [K]', 'Process temperature [K]', 'Rotational speed [rpm]', 'Torque [Nm]', 'Tool wear [min]']

'Split the data into training and testing sets'
train_df, test_df = train_test_split(df, test_size=0.2, random_state=42)

'Create a DoubleMLData object for training'
train_data_dml_base = dml.DoubleMLData(train_df, y_col=outcome, d_cols=treatments, x_cols=covariates)

'Boosted Trees for classification'
boost_class = XGBClassifier()

np.random.seed(123)
dml_irm = dml.DoubleMLIRM(train_data_dml_base, ml_g=boost_class, ml_m=boost_class)
dml_irm.fit(store_predictions=True)

irm_summary = dml_irm.summary

print(irm_summary)
"""

**Comment by SvenKlaassen** [comment]
Hi,

the DoubleMLIRM Model is only available with a single treatment. For multiple treatments you have to fit several models seperately.
Is there a specific reason why you want to use multiple treatments at once (e.g. uniform confidence intervals)?

**Reply by CJames1261** [reply]
Hi,

I just was checking to see if there was a model that could handle multiple treatment values that were binary. in my Use cases I am going to have binary, and continuous values. I'm not sure if there is a model that can handle both types of treatments. 

Currently I have been using the 'DoubleMLPLR' model for my continuous data, but in all reality I'm not really sure which model I should be choosing because there are so many options. I'm not sure if you have a suggestions for a model to use with continuous treatments/binary treatments?

 Would I need to create two separate DoubleMLPLR models where one contains the continuous treatments and the other contains the binary treatments? I am still relatively new to this so I'm not quite sure when I should choose certain models over others. Thanks!

**Reply by SvenKlaassen** [reply, marked as answer]
Your are correct that the `DoubleMLPLR` odel can handle binary treatments, but you should use two seperate models if you want to use binary and continious treatments. The reason would be that the learner for `ml_m` is either used as a regressor or classifier (see [DoubleMLPLR](https://docs.doubleml.org/dev/api/generated/doubleml.DoubleMLPLR.html)). 
You can just define two `DoubleMLData` objects, where the first just uses the continuous treatments and add the binary treatments to the controls `x`. There you can use a regressor as learner for `ml_m`. Afterwards, use the binary treatments and instead add the continuous treatments to the controls. Then you should use a classifier as learner for `ml_m`.

For binary treatments, you could also use the `DoubleMLIRM` model, but there you would have to use a seperate model for each treatment (same procedure as above). The [DoubleMLIRM](https://docs.doubleml.org/dev/guide/models.html#interactive-regression-model-irm) model imposes a bit less structure (full interactions between treatment and confounders possible), whereas the [DoubleMLPLR](https://docs.doubleml.org/dev/guide/models.html#partially-linear-regression-model-plr) model assumes an additive effect.

I hope this can help with the decision.

**Comment by CJames1261** [comment]
Thank you Mr. Klaassen, I very much appreciate your input.

Currently I have created a function that uses your library to compute the ATE for various datasets, where the outcome and treatment variables could be continuous or binary. My function also includes a method to find the top n treatments if user is not aware of what may be an important factor to include in their treatment list. Would you like to inspect this functionality and potentially add this to your library? Thanks!

**Reply by SvenKlaassen** [reply]
Thank you,

that sounds interesting. I have to discuss this with my colleagues.
One potential pitfall for users might be the problem that different effects might have different sets of confounders (the covariates might have to vary for different treatments).

**Reply by CJames1261** [reply]
Sounds good,if your team is interested I can demo my function to you guys. Have a great weekend!

**Reply by SvenKlaassen** [reply]
Sorry, for the late reply. Can I have a look at your function?

**Reply by CJames1261** [reply]
I have created an ongoing Rshiny repo that incorporates Python capability to perform DML using your library. Here is the link to my repo:
[Double-Machine-Learning-Python-RShiny](https://github.com/ctrivino1/Double-Machine-Learning-Python-RShiny).

Note: the UI is still under construction.

**The function that uses your library can be found in the functions folder under dml.py file, this function is near the bottom of the file and is called 'dml_func'. There are other functions in the py script that are used in the dml_func**

If you see in any errors in how I am using your library please feel free to give me feedback. My app is meant for production use, and I am going to eventually create another tab where a user can input their csv and perform causal analysis. 

My function is also able to handle categorical values by dynamically making the string values into binary columns.

Please let me know if you have any questions, thanks!

**Reply by SvenKlaassen** [reply]
Thank you. Looks very interesting. 
I just think that this might be not always completely correct e.g. in the example https://docs.doubleml.org/dev/workflow/workflow.html#problem-formulation conditioning on participation might open a backdoor path if one would like to estimate the effect of income (conditioning on a collider). Therefore it might not always be the best option to exchange treatments $D$ and confounders $X$.

**Reply by CJames1261** [reply]
Hi,

I appreciate the feedback, I think I understand what you are saying. Do you have a suggestion on how I may be able to account for this? Do you think giving a user an option on what covariates to use in the model would help address this?

Would this project be something that you and your team may be interested in having some further discussion about?

**Reply by CJames1261** [reply]
As I have been incorporating your package with my work projects I have found some interesting insights and have been invited to  a conference to present your method/package and my findings this June. Any help to make this as accurate as possible would be much appreciated.

**Reply by SvenKlaassen** [reply]
I think the only option to account for this would be either to give the option on which covariates to use for each treatment or to give a DAG and infer the correponding adjustment sets for each "treatment".

Currently, our team is quite occupied, such that we do not have the capacity for such a project, but you can ask questions at any time and I (or some other team member) will try to answer them.
And any feedback for us is also helpful.

**Reply by CJames1261** [reply]
Thank you for your feedback it is much appreciated. I will be sure to look into how I can address this issue.

If I have any feedback I will be sure to send it your way.

Thank you for your time!
